### PR TITLE
Add configuration

### PR DESCRIPTION
**Pull request by darkvergus**
Added basic configuration still missing how to get the ip address to work so i couldn't get it to work atm though its in the config

to create a new section for the configuration such as xp rates all its required is to create a file like the example i used for ServerConfig and extend ConfigBase

image

After that just add the config to the configuration file like so

image

and you're ready to use it.

ATM it just supports floats, ints, bools and strings but feel free to add more types if you need.

### CODE REVIEW

**Review by AngeloTadeucci — APPROVED** [review]

File: MapleServer2/Config/ConfigManager.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        }
+
+        public static Configuration LoadFromIni(Stream iniStream)
+        {
+            using (StreamReader iniReader = new StreamReader(iniStream))
+            {
+                FileIniDataParser parser = new FileIniDataParser();
+                IniData configdata = parser.ReadData(iniReader);
+
+                Configuration conf = new Configuration();
```
Comment (AlanMorel) [inline MapleServer2/Config/ConfigManager.cs:66]: Looks like lines 43-57 and 66-78 are identical, could probably make a shared helper method and call that instead

**Review by AlanMorel — CHANGES_REQUESTED** [review]

**Comment by rhydev** [comment]
If we're going to add config we should use config files.

File: MapleServer2/Config/ConfigHandler.cs [hunk 2]
```diff
@@ -0,0 +1,199 @@
...
+        {
+            this.commentDelimiter = commentDelimiter;
+            this.file = file;
+        }
+
+        public void Init(string value)
+        {
+            file = null;
+            dictionary.Clear();
+            if (File.Exists(value))
```
Comment (AlanMorel) [inline MapleServer2/Config/ConfigHandler.cs:38]: Maybe do an early return if the file doesn't exist by inverting this if statement

File: MapleServer2/Config/ConfigHandler.cs [hunk 3]
```diff
@@ -0,0 +1,199 @@
...
+            {
+                file = value;
+                using (StreamReader sr = new StreamReader(file))
+                {
+                    string line, section = "";
```
Comment (AlanMorel) [inline MapleServer2/Config/ConfigHandler.cs:43]: Can this be on 2 separate lines?

File: MapleServer2/Config/ConfigHandler.cs [hunk 4]
```diff
@@ -0,0 +1,199 @@
...
+            {
+                if (instance == null)
+                {
+                    instance = new ConfigHandler("config.ini");
+                }
+                return instance;
+            }
+        }
+
+        public string commentDelimiter { get; set; }
```
Comment (AlanMorel) [inline MapleServer2/Config/ConfigHandler.cs:23]: Does this need to be public?

**Review by AlanMorel — CHANGES_REQUESTED** [review]

File: MapleServer2/Config/ConfigHandler.cs [hunk 5]
```diff
@@ -0,0 +1,199 @@
...
+
+            if (value < minValue)
+            {
+                value = minValue;
+            }
+
+            if (value > maxValue)
+            {
+                value = maxValue;
+            }
```
Comment (AngeloTadeucci) [inline MapleServer2/Config/ConfigHandler.cs:137]: Could just use Math.Min and Math.Min

File: MapleServer2/Config/ConfigHandler.cs [hunk 6]
```diff
@@ -0,0 +1,199 @@
...
```
Comment (AngeloTadeucci) [inline MapleServer2/Config/ConfigHandler.cs:162]: same thing here

**Review by AngeloTadeucci — CHANGES_REQUESTED** [review]

**Pull request closed without merging**